libtenzir/builtins/aggregation-functions/mean.cpp

```diff
@@ -0,0 +1,111 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/aggregation_function.hpp>
+#include <tenzir/plugin.hpp>
+
+namespace tenzir::plugins::mean {
+
+namespace {
+
+template <basic_type Type>
+class mean_function final : public aggregation_function {
+public:
+  explicit mean_function(type input_type) noexcept
+    : aggregation_function(std::move(input_type)) {
+    // nop
+  }
+
+private:
+  auto output_type() const -> type override {
+    return type{double_type{}};
+  }
+
+  auto add(const data_view& view) -> void override {
+    using view_type = tenzir::view<type_to_data_t<Type>>;
+    if (caf::holds_alternative<caf::none_t>(view)) {
+      return;
+    }
+    const auto x = static_cast<double>(caf::get<view_type>(view));
+    if constexpr (std::is_same_v<Type, double_type>) {
+      if (std::isnan(x)) {
+        return;
+      }
+    }
+    count_ += 1;
+    mean_ += (x - mean_) / count_;
```

Comment:
Is this done typically like that? I could imagine that tracking the sum has overflow issues, which is why you resorted to this approach.

Reply:
That's just an implementation of Welford's algorithm. It's what you'd typically use to avoid overflows.